packages/lit-dev-content/site/docs/ssr/lifecycle.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Lit SSR lifecycle
+eleventyNavigation:
+  key: Lifecycle
+  parent: Server rendering
+  order: 5
+---
+
+{% labs-disclaimer %}
+
+The tables below lists the standard custom element and Lit element lifecycle methods and whether they are called during SSR and during hydration.
+
+Be mindful that methods called on the server should not contain references to browser/DOM APIs that have not been shimmed. Methods that are not called server-side may contain those references without throwing.
```

Comment:
I think the table reads better now with one column, but now I'm thinking about the iconography.

A main takeaway we want users to have is that they need to be careful in the "called on server" methods, and generally don't need to worry about the "not called on server" methods. But the color/sentiment of the icons is sorta opposite: the "danger" ones are green and the "chill" ones are red.

Not entirely sure how to fix that. I like noting exactly what is run on the server as you have. But when their eye catches the checkmark, I want them to think "be careful." Could add a "Avoid DOM access on server" to every "called on server" method? That's obviously redundant, but maybe hammers it home?

Anyone got any bright ideas?

Reply:
Instead of ✅ and ❌, perhaps we just do "Yes" and "No"? That removes the color. Maybe an * next to all "Yes" with a note of "Avoid DOM access on server"?

Reply:
It's way too hard to tell between Yes and No on their own. Slightly better with the asterisk.
<img width="797" alt="image" src="https://user-images.githubusercontent.com/40413829/185722249-07fc7059-1f10-4280-8697-ef9221c1cbf9.png">
Maybe bold?
<img width="779" alt="image" src="https://user-images.githubusercontent.com/40413829/185722305-b03e608d-147c-4c29-bdd2-6ccca5612916.png">

Reply:
@arthurevans what are your thoughts on this?

Reply:
I think what would be good would be to move this note to a warning box _above_ the tables:

> ⚠️ Methods called on the server should not contain references to browser/DOM APIs that have not been shimmed. Methods that are not called server-side may contain those references without causing breakages.

And then use the ⚠️ icon next to "Yes" in the table, ala "⚠️ Yes":

Reply:
I added the warning as a warn type aside which has a different icon than the emoji. I wonder if congruency here is important?

@e111077 if we were to want to add the same icon within the table, how should I add https://github.com/lit/lit.dev/blob/b0975971b839994e0528baf38acbe0b8c6afae24/packages/lit-dev-content/src/icons/yellow-bang-icon.ts this icon into the markdown file?

Reply:
I'm not 100% sold on the ⚠️ icon, since it still looks like a "hazzard". I'd like to see this framed as not bad that these are called on the server, just a fact. I think we can iterate on this over time and wouldn't block on this.

I do think that the icon should go after the text so the text left aligns properly.

Reply:
I moved it to be after the word.

This is tricky because it's not something users should be afraid of but still be aware of. Something attention grabbing but neutral? We got here because previously @kevinpschaaf pointed out that ✅ seemed too positive when users need to be careful about what goes in there.